extensions-core/kinesis-indexing-service/src/main/java/org/apache/druid/indexing/kinesis/supervisor/KinesisSupervisor.java

```diff
@@ -416,6 +424,52 @@ protected boolean supportsPartitionExpiration()
     return true;
   }
 
+  @Override
+  protected boolean shouldSkipIgnorablePartitions()
+  {
+    return spec.getSpec().getTuningConfig().isSkipIgnorableShards();
+  }
+
+  /**
+   * A kinesis shard is considered to be an ignorable partition if it is both closed and empty
+   * @return set of shards ignorable by kinesis ingestion
+   */
+  @Override
+  protected Set<String> getIgnorablePartitionIds()
```

Comment:
nit - this should be called computeIgnorablePartitionIds() or loadIgnorablePartitionIds()

Reply:
the current verb indicates that it is just a getter but behind the scenes it can do network calls etc to fetch the partition ids.

Reply:
done